src/bootstrap/lib.rs

```diff
@@ -199,6 +199,15 @@ const EXTRA_CHECK_CFGS: &[(Option<Mode>, &'static str, Option<&[&'static str]>)]
     (Some(Mode::Std), "no_global_oom_handling", None),
     (Some(Mode::Std), "freebsd12", None),
     (Some(Mode::Std), "backtrace_in_libstd", None),
+    /* Extra values not defined in the built-in targets yet, but used in std */
+    (Some(Mode::Std), "target_env", Some(&["libnx"])),
+    (Some(Mode::Std), "target_os", Some(&["tvos", "watchos"])),
```

Comment:
compiler/rustc_target/src/spec/aarch64_apple_tvos.rs seems to define tvos in a built-in target, and watchos appears to not be used anywhere in std?